tests/unit/src/test/scala/tests/OnTypeFormattingSuite.scala

```diff
@@ -129,8 +130,38 @@ class OnTypeFormattingSuite extends BaseLspSuite("onTypeFormatting") {
     s"""
        |object Main {
        |  val number = 102
-       |  val str = s"|
-       |  $$number".stripMargin
+       |  val str = s"|" +
+       |    s"$$number".stripMargin
+       |}""".stripMargin
+  )
+
+  check(
```

Comment:
Let's add a couple of test cases with escaped quotes and other symbols to make sure the new algorithm is working.

Reply:
ok